File: capacity-club-api/src/domain-modules/member-plan/model/payload/member-plan-create.payload.ts
import {
  IsOptional,
  IsNotEmpty,
  Length,
  IsString,
  IsNumber,
} from 'class-validator';

import { ApiProperty } from '@nestjs/swagger';
import { memberPlanProperties } from '@common/documentation';
import { ApiCodeResponse } from '@common/api';

/**
 * Class used to create member plan.
 * This class defines the payload used when creating new member plan.
 */
export class MemberPlanCreatePayload {
  @ApiProperty({
    name: memberPlanProperties.title.name,
    description: memberPlanProperties.title.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.MEMBER_PLAN_TITLE_IS_MISSING,
  })
  @Length(1, 80, {
    message: ApiCodeResponse.MEMBER_PLAN_TITLE_LENGTH_ERROR,
  })
  title: string;

  @ApiProperty({
    name: memberPlanProperties.description.name,
    description: memberPlanProperties.description.description,
  })
  @IsString({
    message: ApiCodeResponse.MEMBER_PLAN_DESCRIPTION_INVALID,
  })
  @IsOptional()
  description: string;

  @ApiProperty({
    name: memberPlanProperties.nb_individual_training.name,
    description: memberPlanProperties.nb_individual_training.description,
  })
  @IsNumber()
  @IsOptional()
  nb_individual_training: number;

  @ApiProperty({
    name: memberPlanProperties.nb_collective_training.name,
    description: memberPlanProperties.nb_collective_training.description,
  })
  @IsNumber()
  @IsOptional()
  nb_collective_training: number;

  @ApiProperty({
    name: memberPlanProperties.price.name,
    description: memberPlanProperties.price.description,
  })
  @IsNumber()
  @IsNotEmpty({
    message: ApiCodeResponse.MEMBER_PLAN_PRICE_IS_MISSING,
  })
  price: number;
}

File: capacity_club_mobile_app/lib/domain-features/data/member-plan/model/payload/member_plan_create_payload.dart
import 'package:equatable/equatable.dart';

class MemberPlanCreatePayload extends Equatable {
  final String title;
  final String description;
  final int nb_individual_training;
  final int nb_collective_training;
  final double price;

  const MemberPlanCreatePayload({
    required this.title,
    required this.description,
    required this.nb_individual_training,
    required this.nb_collective_training,
    required this.price,
  });

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'nb_individual_training': nb_individual_training,
        'nb_collective_training': nb_collective_training,
        'price': price,
      };

  @override
  List<Object?> get props => [
        title,
        description,
        nb_individual_training,
        nb_collective_training,
        price,
      ];
}
